Convert FileSetModel modal to a Dialog component

bmquinn opened the pull request:
# Summary 

The modal window is currently being vertically cut off after a certain height, obscuring the "close" button. This PR converts the `FileSetModal` to a `Dialog` component which does a better job of handling the z-index and scrolling issues.

![add_fileset_dialog](https://github.com/nulib/meadow/assets/1395707/1a7efe93-3f6f-48fd-a1f7-1a67ff1b0d24)

Also fixes a bug with the S3ObjectPicker when selecting an S3 object from the list (the behavior in the screenshot is visible on Meadow staging now, this commit makes it go away 😄 )

![SCR-20240627-ovji](https://github.com/nulib/meadow/assets/1395707/ddaafd51-1692-4ff3-80ca-d31546d040b2)


# Specific Changes in this PR
- Updates `FileSetModal.jsx`
- Fixes a small bug with the `S3ObjectPicker` component (see screenshot above)

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.

- [x] Patch
- [ ] Minor
- [ ] Major

# Steps to Test
- Go to the preservation and make sure the `Dialog` component is working properly when you hit the "Add a File Set" button

Also please let developers know if there are any special instructions to test this in the development environment. 

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- Backwards-incompatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- [ ] Requires data migration
- [ ] Requires database triggers disabled during deployment/migration
- [ ] Requires reindex
- [ ] Terraform changes
  - [ ] Adds/requires new or changed Terraform variables
- [ ] Pipeline configuration changes (requires `mix meadow.pipeline.setup` run)
- [ ] Requires new variable added to `miscellany`
- [ ] Specific deployment synchronization instructions with other apps/API's
- [ ] Other specific instructions/tasks


# Tested/Verified
- [ ] End users/stakeholders

kdid commented:
@bmquinn - The close "X" is no longer getting cut off for me. 

I did notice that if I actually click it to close after I've already typed in values in the form, they are not getting cleared. 
Similarly for "cancel" if I've already clicked on a file, and then cancel and try again there's an error showing that wont let you upload saying "Accepted files Uploading NaN undefined"

For the straight shot add/replace w/out any mistakes on the users part everything seems fine.

kdid commented:
@bmquinn - the form field data clearing seems fixed now, but for some reason now there is an extra "Drag 'n' drop a file here, or click to select file" below the option 2, which wasn't there before (and you can't drag and drop there...)

Also, I'm still seeing the "Accepted files Uploading NaN undefined" error that you put a screenshot of in the description if I click on a file name in S3 and then end up canceling or closing. And then try again.

<img width="752" alt="Screenshot 2024-06-28 at 6 20 27 AM" src="https://github.com/nulib/meadow/assets/6372022/86c425b6-c7f9-47f1-be5c-80ea6f37a966">


<img width="716" alt="Screenshot 2024-06-28 at 6 23 37 AM" src="https://github.com/nulib/meadow/assets/6372022/18b9b9b7-a095-4f3e-aeb8-659b80682e76">

mathewjordan commented on app/assets/js/components/Work/Tabs/Preservation/FileSetModal.jsx:
```diff
@@ -189,113 +198,109 @@ function WorkTabsPreservationFileSetModal({
+                  </section>
+
+                  <footer className="modal-card-foot is-justify-content-flex-end">
+                    {s3UploadLocation && (
+                      <>
+                        <Button
+                          isText
+                          type="button"
+                          onClick={handleCancel}
+                          data-testid="cancel-button"
+                        >
+                          Cancel
+                        </Button>
+                        <Button isPrimary type="submit" data-testid="submit-button">
+                          Ingest fileset
+                        </Button>
+                      </>
+                    )}
```
```suggestion
                    <Button
                      isText
                      type="button"
                      onClick={handleCancel}
                      data-testid="cancel-button"
                    >
                      Cancel
                    </Button>
                    <Button
                      disabled={!s3UploadLocation}
                      isPrimary
                      type="submit"
                      data-testid="submit-button"
                    >
                      Ingest fileset
                    </Button>
```

mathewjordan reviewed (approved):
This seems much better with the Radix Dialog! I forgot we had introduced that as a component in Meadow.

Take it or leave it but one thing I noted is that we could use the `disabled` attribute on the button so that the footer is visible at all times, subtly indicating to the user that they have not completed the form yet. My Meadow is being goofy right now and not updating changes in the browser so I couldn't completely verify, but this should be close:

bmquinn commented:
Hi @mathewjordan and @adamjarling, just pushed up a commit that addresses your comments. Made a screencast to show the new `DialogFooter` plus the s3object validation for the buttons.

https://github.com/nulib/meadow/assets/1395707/ba8442d5-7c68-487c-8880-2a78a2452eaf

adamjarling commented on app/assets/js/components/UI/Dialog/Dialog.styled.jsx:
```diff
@@ -43,10 +43,20 @@ const DialogTitle = styled(Dialog.Title, {
   fontWeight: "700",
 });
 
+const DialogFooter = styled(Dialog.Footer, {
```
```suggestion
const DialogFooter = styled("footer", {
```

`Dialog.Footer` is not an export from Radix UI's `Dialog` (https://www.radix-ui.com/primitives/docs/components/dialog#anatomy), so you might run into a compilation error at worst, warning at best.

adamjarling commented on app/assets/js/components/UI/Dialog/Dialog.styled.jsx:
```diff
@@ -43,10 +43,20 @@ const DialogTitle = styled(Dialog.Title, {
+  display: "flex",
+  justifyContent: "flex-end",
+  padding: "1rem",
+  borderTop: "1px solid #e5e7eb",
```
```suggestion

```

adamjarling commented on app/assets/js/components/UI/Dialog/Dialog.styled.jsx:
```diff
@@ -43,10 +43,20 @@ const DialogTitle = styled(Dialog.Title, {
+  justifyContent: "flex-end",
+  padding: "1rem",
+  borderTop: "1px solid #e5e7eb",
+  marginTop: "auto", // Pushes the footer to the bottom of the dialog
```
```suggestion
  marginTop: "1rem",
```

adamjarling reviewed (changes requested):
@bmquinn Sorry for the nitpicky updates, but it's such good work, might as well tighten it up just a tiny bit more.

In addition to the suggestions below, I'd update this style to add a bit of breathing room to the `DialogTitle` via `paddingBottom`.  Maybe something like:

https://github.com/nulib/meadow/blob/0e0a0117ab09bc100bef2ee8c5ea296e1b236d0b/app/assets/js/components/UI/Dialog/Dialog.styled.jsx#L42-L44

```jsx
const DialogTitle = styled(Dialog.Title, {
  fontWeight: "700",
  paddingBottom: "1rem",
});
```

bmquinn commented:
Thanks a ton for your code suggestions @adamjarling 🥳 , I merged them all into one commit. Just tested the branch in my dev environment and everything is looking good on my end. Feel free to squash and merge if you're down with the changes, otherwise I'd be happy to.
